### PR TITLE
Zoom to bookmark extent

### PR DESCRIPTION
**Pull request by benoitblanc**
Hi,

This PR allows user to only zoom to a bookmark extent without changing theme or load/open the bookmark in the current window or new tab.

Thanks for the review

### CODE REVIEW

File: plugins/Bookmark.jsx [hunk 1]
```diff
@@ -18,7 +18,9 @@ import Spinner from '../components/Spinner';
 import {createBookmark, getUserBookmarks, removeBookmark, updateBookmark} from '../utils/PermaLinkUtils';
 import './style/Bookmark.css';
 import isEmpty from 'lodash.isempty';
-
+import { zoomToPoint } from '../actions/map';
+import { resolveBookmark } from '../utils/PermaLinkUtils';
+import MapUtils from '../utils/MapUtils';
```
Comment (manisandro) [inline plugins/Bookmark.jsx:24]: Please keep imports sorted (first absolute, then relative, and lastly the style import)

File: plugins/Bookmark.jsx [hunk 2]
```diff
@@ -118,6 +127,14 @@ class Bookmark extends React.Component {
             location.href = url;
         }
     };
+    zoomToBookmarkExtent = (bookmarkkey) => {
+        resolveBookmark(bookmarkkey, (params) => {
+            const scale = parseFloat(params.s);
+            const zoom = MapUtils.computeZoom(this.props.mapScales, scale);
+            const center = params.c.split(/[;,]/g).map(x => parseFloat(x));
```
Comment (manisandro) [inline plugins/Bookmark.jsx:134]: Depending on `urlPositionFormat` in `config.json`, the url parameters may contain `c` and `s`, or `e` for extent. You might also want to handle the `e` case here.

File: plugins/Bookmark.jsx (hunk 1 above)
Comment (benoitblanc) [reply plugins/Bookmark.jsx:24]: Do we have some eslint rule for that ?

Comment (manisandro) [reply plugins/Bookmark.jsx:24]: Currently not, I suppose we could enable https://eslint.org/docs/latest/rules/sort-imports

Comment (benoitblanc) [reply plugins/Bookmark.jsx:24]: Ok I was looking to the same link. I can take the task ;-)

File: plugins/Bookmark.jsx [hunk 3]
```diff
@@ -121,6 +130,19 @@ class Bookmark extends React.Component {
...
+    zoomToBookmarkExtent = (bookmarkkey) => {
+        resolveBookmark(bookmarkkey, (params) => {
+            if ('c' in params && 's' in params) {
+                const scale = parseFloat(params.s);
+                const zoom = MapUtils.computeZoom(this.props.mapScales, scale);
+                const center = params.c.split(/[;,]/g).map(x => parseFloat(x));
+                this.props.zoomToPoint(center, zoom, this.props.mapCrs);
+            } else if ('e' in params) {
+                const bounds = (params.e).split(',').map(n => parseFloat(n));
+                this.props.zoomToExtent(bounds, this.props.mapCrs);
```
Comment (manisandro) [inline plugins/Bookmark.jsx:142]: Sorry I forgot earlier, there is also the `urlPositionCrs` config option, which will add a `crs` URL parameter if `urlPositionCrs` is different than `mapCrs`. Something like `params.crs ?? this.props.mapCrs` should do.

Comment (benoitblanc) [reply plugins/Bookmark.jsx]: Thanks, I was not aware of these options

**Comment by manisandro** [comment]
Thanks!

**Pull request merged**